packages/client/src/app/shared/auth/services/auth.guard.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, CanActivateChild, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { JwtHelperService } from '@auth0/angular-jwt';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Store } from '@ngxs/store';
+import * as lodash from 'lodash';
+
+import { AuthState } from '../auth.state';
+import { JwtPayload } from '@pulp-fiction/models/auth';
+import { FrontendUser } from '@pulp-fiction/models/users';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthGuard implements CanActivate, CanActivateChild {
+    helper = new JwtHelperService();
+
+    constructor(private store: Store, private snackBar: MatSnackBar) {}
+
+    /**
+     * Verifies that a user can access a protected route. If their JWT is expired, it makes a request to the backend to
+     * verify that an active session is still present. If one is, the route request is approved. If one isn't, then the
+     * request is denied.
```

Comment:
Is the part about requesting the backend here still true?

Reply:
yes